src/xmppjs/XJSGateway.ts

```diff
@@ -163,6 +177,16 @@ export class XmppJsGateway implements IGateway {
                 "groupchat",
             )
         );
+
+        // add the message to the room history
+        const historyStanza = new StzaMessage(
```

Comment:
We should probably respect m.room.history_visibility to ensure that XMPP users are not leaked history btw.

Reply:
It was asked before how to determine the vis setting.
https://github.com/matrix-org/matrix-bifrost/pull/227#issuecomment-769555110

Reply:
Added PRs to hopefully address this.